src/ngResource/resource.js

```diff
@@ -593,7 +601,13 @@ angular.module('ngResource', ['ng']).
               extend({}, extractParams(data, action.params || {}), params),
               action.url);
 
-            var promise = $http(httpConfig).then(function(response) {
+            var promise = $q.resolve(httpConfig);
```

Comment:
Why not directly: `$q.when(requestInterceptor(httpConfig), null, requestErrorInterceptor)`